gvc.go

```diff
@@ -133,21 +130,45 @@ func cleanup(path string, opts options) error {
 		if lastVendorPath == "" {
 			lastVendorPath = localPath
 		}
+		lastVendorPathDir := filepath.Dir(lastVendorPath)
 
 		keep := false
-		for _, name := range pkgList {
-			// If the file's parent directory is a needed package, keep it.
-			if !info.IsDir() && filepath.Dir(lastVendorPath) == name {
-				if opts.onlyCode {
+
+		if info.IsDir() {
+			for _, name := range pkgList {
+				// If a directory is a needed package then keep it
+				if name == lastVendorPath {
+					keep = true
+					break
+				}
+			}
+		} else {
```

Comment:
Perhaps we can avoid an identation removing the `else` and returning nil in the above block.